api-call.py

```diff
@@ -1,29 +1,322 @@
+                                                vehicles_data = vehicles_data_response['data']
+
+                                                if 'attributes' in vehicles_data:
+                                                    vehicles_attributes = vehicles_data['attributes']
+
+                                                    if 'name' in vehicles_attributes and 'heading' in vehicles_attributes:
+                                                        get_vehicle_name = vehicles_attributes['name']
+                                                        get_vehicle_heading = vehicles_attributes['heading']
+
+                                                    else:
+                                                        print("Error: 'name' or 'heading' key is missing in vehicles attributes")
+                                                else:
+                                                    print("Error: 'attributes' key is missing in vehicles data")
+                                            else:
+                                                print("Error: 'data' key is missing in vehicles data response")
+                                        else:
+                                            print("Error: 'vehicleID' or 'eta' key is missing in the first arrival object")
+                                    else:
+                                        print("Error: 'arrivals' key is missing or empty in arrivals attributes")
+                                else:
+                                    print("Error: 'attributes' key is missing or empty in the first arrivals data")
+                            else:
+                                print("Error: 'data' key is missing in arrivals data response")
+                        else:
+                            print("Error: 'stopID' or 'description' key is missing in stop object")
+
+                        print(f"Route ID: {url_obj['route_id']}, Route Name: {route_name}, Stop ID: {stop_id}, Stop Name: {description}, Vehicle Name: {get_vehicle_name}, Vehicle Number: {get_vehicle_id}, Heading: {get_vehicle_heading}, ETA for arrival to Stop: {get_eta_at_stop}")
+
+    def main(self) -> None:
+        """Driver function of class
+        """
+        while True:
+            self.show_tasks()
+            user_choice = self.get_user_choice()
+            url_obj = self.get_url(user_choice)
+            if user_choice == '0':
+                print(url_obj)
+                break
+            if (
+                user_choice in ('1', '2')
+            ):
+                self.get_api_data(
+                    url_obj['url'],
+                    url_obj['parameters'],
+                    url_obj['header']
+                )
+            elif user_choice == '3':
+                self.get_text_books_with_term_date(url_obj)
+            elif user_choice == '4':
+                self.get_stops_vehicles_on_route(url_obj)
```

Comment:
For successful calls among these options, what a user should expect to see when using your script?

Reply:
Menu in an infinite loop helps them access these APIs with this description:

Which API data do you want to check?
0. Quit
1. Beavers Bus
2. Terms
3. Search Text Books of a Term by a custom/any Date
4. Get Details of a Vehicle and its Stops given a specified Route, with ETA and current destination